Userland/Libraries/LibCrypto/Cipher/AES.cpp

```diff
@@ -152,24 +147,267 @@ void AESCipherKey::expand_encrypt_key(ReadonlyBytes user_key, size_t bits)
+    AK::SIMD::store_unaligned(&round_key[0 * 4], ta);
+    tb = __builtin_ia32_aeskeygenassist128(ta, 0x01); ta = AESCipherKey_expand_encrypt_key_128_assist(ta, tb); AK::SIMD::store_unaligned(&round_key[ 1 * 4], ta);
+    tb = __builtin_ia32_aeskeygenassist128(ta, 0x02); ta = AESCipherKey_expand_encrypt_key_128_assist(ta, tb); AK::SIMD::store_unaligned(&round_key[ 2 * 4], ta);
+    tb = __builtin_ia32_aeskeygenassist128(ta, 0x04); ta = AESCipherKey_expand_encrypt_key_128_assist(ta, tb); AK::SIMD::store_unaligned(&round_key[ 3 * 4], ta);
+    tb = __builtin_ia32_aeskeygenassist128(ta, 0x08); ta = AESCipherKey_expand_encrypt_key_128_assist(ta, tb); AK::SIMD::store_unaligned(&round_key[ 4 * 4], ta);
+    tb = __builtin_ia32_aeskeygenassist128(ta, 0x10); ta = AESCipherKey_expand_encrypt_key_128_assist(ta, tb); AK::SIMD::store_unaligned(&round_key[ 5 * 4], ta);
+    tb = __builtin_ia32_aeskeygenassist128(ta, 0x20); ta = AESCipherKey_expand_encrypt_key_128_assist(ta, tb); AK::SIMD::store_unaligned(&round_key[ 6 * 4], ta);
+    tb = __builtin_ia32_aeskeygenassist128(ta, 0x40); ta = AESCipherKey_expand_encrypt_key_128_assist(ta, tb); AK::SIMD::store_unaligned(&round_key[ 7 * 4], ta);
+    tb = __builtin_ia32_aeskeygenassist128(ta, 0x80); ta = AESCipherKey_expand_encrypt_key_128_assist(ta, tb); AK::SIMD::store_unaligned(&round_key[ 8 * 4], ta);
+    tb = __builtin_ia32_aeskeygenassist128(ta, 0x1b); ta = AESCipherKey_expand_encrypt_key_128_assist(ta, tb); AK::SIMD::store_unaligned(&round_key[ 9 * 4], ta);
+    tb = __builtin_ia32_aeskeygenassist128(ta, 0x36); ta = AESCipherKey_expand_encrypt_key_128_assist(ta, tb); AK::SIMD::store_unaligned(&round_key[10 * 4], ta);
+}
+}
+#endif
 
-    round_key = round_keys();
+#if (ARCH(I386) || ARCH(X86_64))
+namespace
+{
+static void AESCipherKey_expand_encrypt_key_192_assist(AK::SIMD::i32x4& a, AK::SIMD::i32x4& b, AK::SIMD::i32x4& c)
+{
+    AK::SIMD::i32x4 ta {};
+    AK::SIMD::i32x4 tb {};
+    AK::SIMD::i32x4 tc {};
+    AK::SIMD::i32x4 td {};
+
+    ta = AK::SIMD::as<AK::SIMD::i32x4>(a);
+    tb = AK::SIMD::as<AK::SIMD::i32x4>(b);
+    tc = AK::SIMD::as<AK::SIMD::i32x4>(c);
+    tb = AK::SIMD::i32x4 { tb[1], tb[1], tb[1], tb[1] };
+    td = AK::SIMD::i32x4 { 0, ta[0], ta[1], ta[2] };
+    ta ^= td;
+    td = AK::SIMD::i32x4 { 0, td[0], td[1], td[2] };
+    ta ^= td;
+    td = AK::SIMD::i32x4 { 0, td[0], td[1], td[2] };
+    ta ^= td;
+    ta ^= tb;
+    tb = AK::SIMD::i32x4 { ta[3], ta[3], ta[3], ta[3] };
+    td = AK::SIMD::i32x4 { 0, tc[0], tc[1], tc[2] };
+    tc ^= td;
+    tc ^= tb;
+    a = AK::SIMD::as<AK::SIMD::i64x2>(ta);
+    b = AK::SIMD::as<AK::SIMD::i64x2>(tb);
+    c = AK::SIMD::as<AK::SIMD::i64x2>(tc);
+}
+}
+#endif
+
+#if (ARCH(I386) || ARCH(X86_64))
+namespace
+{
+static void __attribute__((target("aes"), used)) AESCipherKey_expand_encrypt_key_192(u32* round_key, ReadonlyBytes user_key)
+{
+    union aligned_buf
+    {
+        unsigned char m_buf[16];
+        AK::SIMD::i32x4 m_vec;
+    };
```

Comment:
This seems like it could be done better,
will send a diff on discord

Reply:
OK, used the diff.